modules/ocf_www/files/build-vhosts

```diff
@@ -133,6 +140,14 @@ class VirtualHost(namedtuple('VirtualHost', (
         assert self.is_redirect
         return self.bind_dest
 
+    @property
+    def has_websockets(self):
+        return self.bind_type == 'socket' and len(self.additional_rules.get('ws', [])) > 0
+
+    @property
+    def websocket_locations(self):
+        return self.additional_rules['ws']
```

Comment:
Instead of this and using `has_websockets` for a conditional, why not just have `websocket_locations` return an empty list if it's not applicable and then nothing will show up in the template?

So something like:
```python
@property
def websocket_locations(self):
    return self.additional_rules.get('ws', [])
```
(without the `has_websockets` at all)
Or, if you do want to keep the socket check, maybe this (could perhaps be written in a nicer way):
```
@property
def websocket_locations(self):
    if self.bind_type == 'socket' and 'ws' in self.additional_rules:
        return self.additional_rules['ws']
    else:
        return []
```

And then in the template, you can remove the `has_websockets` check entirely:
```
{% for ws_location in vhost.websocket_locations %}
location /{{ws_location}} {
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_set_header Host $host;
        proxy_pass http://unix:{{vhost.socket}};
}
{% endfor %}
```